### PR TITLE
[DEV-2607] Extend tunnels with name field

### PR DESCRIPTION
**Pull request by morestatic**
Added the name field to the active tunnel model (called Remote)

Notes:
- Refactored the ClientService into the ClientService interface and a ClientServiceProvider implementation
- Add API unit test to check for the tunnel name field (using a mock based on the ClientService interface)

### CODE REVIEW

**Review by thorstenkramm — APPROVED** [review]
@domenipavec Please have a look as well.

File: server/api_handler_clients_test.go [hunk 1]
```diff
@@ -343,3 +350,100 @@ func TestGetCorrespondingSortFuncError(t *testing.T) {
...
+
+func TestHandlePutTunnelWithName(t *testing.T) {
+	connMock := test.NewConnMock()
+	connMock.ReturnOk = true
+	connMock.ReturnResponsePayload = []byte("{ \"IsAllowed\": true }")
+
+	testCases := []struct {
+		Name           string
+		URL            string
+		TunnelName     string
```
Comment (domenipavec) [inline server/api_handler_clients_test.go:390]: This doesn't seem to be used.

File: server/server.go [hunk 2]
```diff
@@ -309,6 +309,10 @@ func (s *Server) Close() error {
 	return wg.Wait()
 }
 
+func (s *Server) GetClientService() ClientService {
```
Comment (domenipavec) [inline server/server.go:312]: This is not used?

**Review by domenipavec — DISMISSED** [review]

File: server/server.go (hunk 2 above)
Comment (morestatic) [reply server/server.go:312]: ah good catch. yes, not required. i've removed it.

File: server/api_handler_clients_test.go (hunk 1 above)
Comment (morestatic) [reply server/api_handler_clients_test.go:390]: I can't quite see (or not quite getting) what you're referencing. But I can see that the ExpectedStatus isn't really being used. Is that what you mean? I'll tidy the ExpectedStatus up but let me know if something else? The use of a test table is borderline. Do you mean that?

Comment (domenipavec) [reply server/api_handler_clients_test.go:390]: I meant that `TunnelName` property is set in `testCases` but not used in the actual tests.

Comment (morestatic) [reply server/api_handler_clients_test.go:390]: you're right. i really couldn't see that one. i think for a while i was going to Sprintf it into the strings but decided that was over the top. ok. removed and also removed the ExpectedStatus use.

**Review by thorstenkramm — APPROVED** [review]

**Pull request merged**